packages/core/helper-plugin/src/utils/getDisplayName/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Retrieves the display name of an admin panel user
+ * @typedef AdminUserNamesAttributes
+ * @property {string} firstname
+ * @property {string} lastname
+ * @property {string} username
+ * @property {string} email
+ *
+ * @type {(user: AdminUserNamesAttributes, formatMessage: import('react-intl').formatMessage) => string}
+ */
+const getDisplayName = ({ firstname, lastname, username, email }, formatMessage) => {
+  if (username) {
+    return username;
+  }
+
+  // firstname is not required if the user is created with a username
+  if (firstname) {
+    return formatMessage(
+      {
+        id: 'global.fullname',
+        defaultMessage: '{firstname} {lastname}',
```

Comment:
Haven't tested it, but I think we can use the [`select`](https://formatjs.io/docs/core-concepts/icu-syntax#select-format) syntax to render the whitespace conditionally and avoid the `.trim()`

```suggestion
        defaultMessage: '{firstname}{lastname, select, undefined {} other { {lastname}}}',
```

Reply:
apparently lastname is an empty string when it does not exist, I would say it could be a bit more work to check for possible lastname null or empty values. What do you think if we leave the trim?

Reply:
Yeah, if that can happen I am in favor of that. Since you know me you probably know I'd love to find out why it is empty string over null, but ok :D

Reply:
FYI On 1st admin registration the client sends empty string instead of no value or null for the lastname when you leave the input empty and the BE allows it

Reply:
Wonderful, thanks. I'll create a bugreport.

Reply:
https://github.com/strapi/strapi/issues/17287